runtime/compiler/p/codegen/J9PPCSnippet.cpp

```diff
@@ -1879,6 +1879,17 @@ uint32_t TR::getCCPreLoadedCodeSize()
 #define CCEYECATCHER(a, b, c, d) (((a) << 24) | ((b) << 16) | ((c) << 8) | ((d) << 0))
 #endif
 
+static void performCCPreLoadedBinaryEncoding(uint8_t *buffer, TR::CodeGenerator *cg)
+   {
+   cg->setBinaryBufferStart(buffer);
+   cg->setBinaryBufferCursor(buffer);
+   for (TR::Instruction *i = cg->getFirstInstruction(); i != NULL; i = i->getNext())
+      {
+      i->estimateBinaryLength(cg->getBinaryBufferCursor() - cg->getBinaryBufferStart());
+      cg->setBinaryBufferCursor(i->generateBinaryEncoding());
```

Comment:
There is only one BinaryBufferStart value (at least for a logical encoding unit such as a specific compilation). The way i->estimateBinaryLength is done looks wrong to me.  The next instruction will see longer estimated length. Reversing the two lines above and fixing using cg->getBinaryBufferStart would make more senses.

Reply:
I think you're confused about what this code is actually doing... In this case, `estimateBinaryLength` is the method of `TR::Instruction` in charge of _generating_ the binary length estimate and it takes in the current estimated offset into the binary buffer.

In this case, using the actual offset instead of keeping track of an estimate is fine since we don't ever actually use the estimated positions we generate here outside of verifying that the estimates are not violated. It's assumed that all conditional branches within a per-CC helper snippet are within range and the new binary encoder will fail with an assertion if that assumption doesn't hold and an out-of-range branch is encountered.